DecipherNow/gm-fabric-dashboard
# ci: generate draft GitHub releases

Resolves #1552 

## Proposed Changes

  - Modify CircleCI config to have distinct workflows for tagged git releases with a tag starting with `v` and 
 normal branches.
  - Generate GitHub release drafts automatically when a git release is tagged started with a 'v' and it builds cleanly and passes e2e tests
  
## How to Test

  - Review the code
  - Fork this branch locally into a test branch, make a change, and push to origin to make sure unit testing and coverage works as expected
  - Run `git tag -a v1.0.6-alpha9 -m "Grey Matter Fabric Dashboard version 1.0.6-alpha9"` and `git push origin --tags` to make sure build, e2e tests, and a draft GitHub release is created as expected.
- Delete the draft GitHub release you created.
